File: app/src/main/java/techies/it/preschoolmaths/ui/GameLevel2.java
package techies.it.preschoolmaths.ui;

import android.annotation.SuppressLint;
import android.annotation.TargetApi;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffXfermode;
import android.graphics.Typeface;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import butterknife.Bind;
import butterknife.ButterKnife;
import techies.it.preschoolmaths.R;
import techies.it.preschoolmaths.utils.DrawingView;

public class GameLevel2 extends AppCompatActivity implements View.OnClickListener
{
    private int currentApiVersion;

    DrawingView drawingView;

    private Paint mPaint;

    private boolean mBtnEnable = true;

    int apple = 0;

    @Bind(R.id.contentLayoutGameLevel2)
    RelativeLayout mContentLayoutGameLevel2;

    @Bind(R.id.backBT)
    ImageButton mBack;

    @Bind(R.id.smallBoardLabel)
    TextView smallBoardLabel;

    @Bind(R.id.bigBoardLabel)
    TextView bigBoardLabel;

    @Bind(R.id.star1)
    ImageView mStar1;

    @Bind(R.id.star2)
    ImageView mStar2;

    @Bind(R.id.star3)
    ImageView mStar3;

    @Bind(R.id.star4)
    ImageView mStar4;

    @Bind(R.id.star5)
    ImageView mStar5;

    @Bind(R.id.star6)
    ImageView mStar6;

    @Bind(R.id.star7)
    ImageView mStar7;

    @Bind(R.id.star8)
    ImageView mStar8;

    @Bind(R.id.star9)
    ImageView mStar9;

    @Bind(R.id.cloud0IV)
    ImageView mCloud0;

    @Bind(R.id.cloud6IV)
    ImageView mCloud6;

    @Bind(R.id.cloud7IV)
    ImageView mCloud7;

    @Bind(R.id.cloud1IV)
    ImageView mCloud1;

    @Bind(R.id.cloud3IV)
    ImageView mCloud3;

    @Bind(R.id.sunRaysIV)
    ImageView mSunRays;

    Animation animation270Sec, animation330Sec, animation390Sec, rotation;

    @Bind(R.id.dusterIB)
    ImageButton mDuster;

    @Bind(R.id.chalkIB)
    ImageButton mChalk;

    @Bind(R.id.btNext)
    ImageButton mNext;

    @Bind(R.id.btPrevious)
    ImageButton mPrevious;


    @TargetApi(Build.VERSION_CODES.KITKAT)
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        currentApiVersion = android.os.Build.VERSION.SDK_INT;

        final int flags = View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_FULLSCREEN
                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;

        // This work only for android 4.4+
        if (currentApiVersion >= Build.VERSION_CODES.KITKAT)
        {

            getWindow().getDecorView().setSystemUiVisibility(flags);

            // Code below is to handle presses of Volume up or Volume down.
            // Without this, after pressing volume buttons, the navigation bar will
            // show up and won't hide
            final View decorView = getWindow().getDecorView();
            decorView
                    .setOnSystemUiVisibilityChangeListener(new View.OnSystemUiVisibilityChangeListener()
                    {

                        @Override
                        public void onSystemUiVisibilityChange(int visibility)
                        {
                            if ((visibility & View.SYSTEM_UI_FLAG_FULLSCREEN) == 0)
                            {
                                decorView.setSystemUiVisibility(flags);
                            }
                        }
                    });
        }
        else
        {
            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        }

        setContentView(R.layout.activity_game_level2);
        ButterKnife.bind(this);

        mPrevious.setEnabled(false);
        mPrevious.setClickable(false);

        Typeface typeface = Typeface.createFromAsset(getAssets(), "font/cabin_sketch_bold.ttf");


        smallBoardLabel.setTypeface(typeface);


        mPaint = new Paint();
        mPaint.setAntiAlias(true);
        mPaint.setDither(true);
        mPaint.setColor(Color.WHITE);
        mPaint.setStyle(Paint.Style.STROKE);
        mPaint.setStrokeJoin(Paint.Join.ROUND);
        mPaint.setStrokeCap(Paint.Cap.ROUND);
        GameLevel1.brushSize(GameLevel2.this, mPaint);

        drawingView = new DrawingView(this, mPaint);
        drawingView.setBackgroundColor(Color.TRANSPARENT);
        drawingView.setButtonEnableListener(new DrawingView.BtnEnableListener()
        {
            @Override
            public void enableButton(boolean enable)
            {
                mBtnEnable = enable;
            }
        });
        mContentLayoutGameLevel2.addView(drawingView);



        int width = 0;
        int height = 0;
        if (getResources().getBoolean(R.bool.isTab))
        {
            // for tab
            bigBoardLabel.setTypeface(typeface);

            animation270Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_270_sec_600);

            animation330Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_330_sec_600);

            animation390Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_390_sec_600);

            rotation = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.rotation_600);

            mCloud0.startAnimation(animation270Sec);
            mCloud1.startAnimation(animation330Sec);
            mCloud3.startAnimation(animation330Sec);
            mCloud6.startAnimation(animation390Sec);
            mCloud7.startAnimation(animation390Sec);
            mSunRays.startAnimation(rotation);


        }
        else
        {
            // for mobile
            animation270Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_270_sec_320);

            animation330Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_330_sec_320);

            animation390Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_390_sec_320);

            rotation = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.rotation_320);

            mCloud0.startAnimation(animation270Sec);
            mCloud1.startAnimation(animation330Sec);
            mCloud3.startAnimation(animation330Sec);
            mCloud6.startAnimation(animation390Sec);
            mCloud7.startAnimation(animation390Sec);
            mSunRays.startAnimation(rotation);


        }

        mChalk.setOnClickListener(this);
        mDuster.setOnClickListener(this);
        mNext.setOnClickListener(this);
        mPrevious.setOnClickListener(this);
        mBack.setOnClickListener(this);
    }

    @SuppressLint("NewApi")
    @Override
    public void onWindowFocusChanged(boolean hasFocus)
    {
        super.onWindowFocusChanged(hasFocus);
        if (currentApiVersion >= Build.VERSION_CODES.KITKAT && hasFocus)
        {
            getWindow().getDecorView().setSystemUiVisibility(
                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
        }
    }

    @Override
    public void onClick(View v)
    {
        if (mBtnEnable)
        {
            mPaint.setXfermode(null);
            mPaint.setAlpha(0xFF);

            if (v.getId() == R.id.btNext)
            {
                Log.d("Next: ", String.valueOf(apple));

                apple++;

                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                GameLevel1.brushSize(GameLevel2.this, mPaint);

                switch (apple)
                {
                    case 1:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);
                        mPrevious.setEnabled(true);
                        mPrevious.setClickable(true);
                        mPrevious.setAlpha(1.0f);
                        mNext.setClickable(true);
                        mNext.setAlpha(1.0f);
                        break;
                    case 2:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.INVISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.INVISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 3:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);

                        break;
                    case 4:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);

                        break;
                    case 5:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 6:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.INVISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.INVISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 7:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 8:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.VISIBLE);
                        mNext.setClickable(false);
                        mNext.setEnabled(false);
                        mNext.setAlpha(0.5f);

                        break;

                }
            }
            else if (v.getId() == R.id.btPrevious)
            {
                apple--;

                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                GameLevel1.brushSize(GameLevel2.this, mPaint);

                switch (apple)
                {
                    case 0:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.INVISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.INVISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);
                        mPrevious.setClickable(false);
                        mPrevious.setEnabled(false);
                        mPrevious.setAlpha(0.5f);

                        break;
                    case 1:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);

                        break;
                    case 2:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.INVISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.INVISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 3:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);

                        break;
                    case 4:
                        mStar1.setVisibility(View.INVISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.INVISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.INVISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.INVISIBLE);

                        break;
                    case 5:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.INVISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.INVISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 6:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.INVISIBLE);
                        mStar5.setVisibility(View.VISIBLE);
                        mStar6.setVisibility(View.INVISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.VISIBLE);

                        break;
                    case 7:
                        mStar1.setVisibility(View.VISIBLE);
                        mStar2.setVisibility(View.VISIBLE);
                        mStar3.setVisibility(View.VISIBLE);
                        mStar4.setVisibility(View.VISIBLE);
                        mStar5.setVisibility(View.INVISIBLE);
                        mStar6.setVisibility(View.VISIBLE);
                        mStar7.setVisibility(View.VISIBLE);
                        mStar8.setVisibility(View.VISIBLE);
                        mStar9.setVisibility(View.VISIBLE);
                        mPrevious.setClickable(true);
                        mPrevious.setEnabled(true);
                        mPrevious.setAlpha(1.0f);
                        mNext.setClickable(true);
                        mNext.setEnabled(true);
                        mNext.setAlpha(1.0f);

                        break;


                }
            }
            else if (v.getId() == R.id.dusterIB)
            {
                mPaint.setStrokeWidth(40);
                mPaint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.CLEAR));
            }
            else if (v.getId() == R.id.chalkIB)
            {
                mPaint.setColor(Color.WHITE);
                GameLevel1.brushSize(GameLevel2.this, mPaint);
            }
            else if (v.getId() == R.id.backBT)
            {
                finish();
            }
        }
    }
}

File: app/src/main/java/techies/it/preschoolmaths/ui/GameLevel2.kt
package techies.it.preschoolmaths.ui

import android.annotation.SuppressLint
import android.graphics.*
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import techies.it.preschoolmaths.R
import techies.it.preschoolmaths.databinding.ActivityGameLevel2Binding
import techies.it.preschoolmaths.ui.GameLevel1.Companion.brushSize
import techies.it.preschoolmaths.utils.DrawingView

class GameLevel2 : AppCompatActivity(), View.OnClickListener {

    private lateinit var binding: ActivityGameLevel2Binding
    var view: View?=null
    var drawingView: DrawingView? = null
    private var mPaint: Paint? = null
    private var mBtnEnable = true
    var apple = 0
    var animation270Sec: Animation? = null
    var animation330Sec: Animation? = null
    var animation390Sec: Animation? = null
    var rotation: Animation? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityGameLevel2Binding.inflate(layoutInflater)
        view = binding.root
        setContentView(view)

        binding.btPrevious.isEnabled = false
        binding.btPrevious.isClickable = false
        
        mPaint = Paint()
        mPaint?.isAntiAlias = true
        mPaint?.isDither = true
        mPaint?.color = Color.WHITE
        mPaint?.style = Paint.Style.STROKE
        mPaint?.strokeJoin = Paint.Join.ROUND
        mPaint?.strokeCap = Paint.Cap.ROUND
        brushSize(this@GameLevel2, mPaint)
        drawingView = DrawingView(this, mPaint)
        drawingView?.setBackgroundColor(Color.TRANSPARENT)
        drawingView?.setButtonEnableListener { enable -> mBtnEnable = enable }
        binding.contentLayoutGameLevel2.addView(drawingView)
        val width = 0
        val height = 0
        if (resources.getBoolean(R.bool.isTab)) {
            // for tab
            animation270Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_270_sec_600
            )
            animation330Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_330_sec_600
            )
            animation390Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_390_sec_600
            )
            rotation = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.rotation_600
            )
            binding.cloud0IV.startAnimation(animation270Sec)
            binding.cloud1IV.startAnimation(animation330Sec)
            binding.cloud3IV.startAnimation(animation330Sec)
            binding.cloud6IV.startAnimation(animation390Sec)
            binding.cloud7IV.startAnimation(animation390Sec)
            binding.sunRaysIV.startAnimation(rotation)
        } else {
            // for mobile
            /*animation270Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_270_sec_320
            )
            animation330Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_330_sec_320
            )
            animation390Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_390_sec_320
            )*/
            rotation = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.rotation_320
            )
            binding.sunRaysIV.startAnimation(rotation)
        }
        
        binding.chalkIB.setOnClickListener(this)
        binding.dusterIB.setOnClickListener(this)
        binding.btNext.setOnClickListener(this)
        binding.btPrevious.setOnClickListener(this)
        binding.backBT.setOnClickListener(this)
    }

    @SuppressLint("NewApi")
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.decorView.windowInsetsController?.hide(
                android.view.WindowInsets.Type.statusBars()
                        or android.view.WindowInsets.Type.navigationBars()
            )
        }else{
            WindowCompat.setDecorFitsSystemWindows(window, false)
            view?.let {
                WindowInsetsControllerCompat(window, it).let { controller ->
                    controller.hide(WindowInsetsCompat.Type.systemBars())
                    controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
                }
            }
        }
    }

    override fun onClick(v: View) {
        if (mBtnEnable) {
            mPaint?.xfermode = null
            mPaint?.alpha = 0xFF
            if (v.id == R.id.btNext) {
                Log.d("Next: ", apple.toString())
                apple++
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel2, mPaint)
                when (apple) {
                    1 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.INVISIBLE
                        binding.btPrevious.isEnabled = true
                        binding.btPrevious.isClickable = true
                        binding.btPrevious.alpha = 1.0f
                        binding.btNext.isClickable = true
                        binding.btNext.alpha = 1.0f
                    }
                    2 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.INVISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.INVISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    3 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.INVISIBLE
                    }
                    4 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.INVISIBLE
                    }
                    5 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    6 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.INVISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.INVISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    7 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    8 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.VISIBLE
                        binding.btNext.isClickable = false
                        binding.btNext.isEnabled = false
                        binding.btNext.alpha = 0.5f
                    }
                }
            } else if (v.id == R.id.btPrevious) {
                apple--
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel2, mPaint)
                when (apple) {
                    0 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.INVISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.INVISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.INVISIBLE
                        binding.btPrevious.isClickable = false
                        binding.btPrevious.isEnabled = false
                        binding.btPrevious.alpha = 0.5f
                    }
                    1 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.INVISIBLE
                    }
                    2 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.INVISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.INVISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    3 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.INVISIBLE
                    }
                    4 -> {
                        binding.star1.visibility = View.INVISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.INVISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.INVISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.INVISIBLE
                    }
                    5 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.INVISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.INVISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    6 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.INVISIBLE
                        binding.star5.visibility = View.VISIBLE
                        binding.star6.visibility = View.INVISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.VISIBLE
                    }
                    7 -> {
                        binding.star1.visibility = View.VISIBLE
                        binding.star2.visibility = View.VISIBLE
                        binding.star3.visibility = View.VISIBLE
                        binding.star4.visibility = View.VISIBLE
                        binding.star5.visibility = View.INVISIBLE
                        binding.star6.visibility = View.VISIBLE
                        binding.star7.visibility = View.VISIBLE
                        binding.star8.visibility = View.VISIBLE
                        binding.star9.visibility = View.VISIBLE
                        binding.btPrevious.isClickable = true
                        binding.btPrevious.isEnabled = true
                        binding.btPrevious.alpha = 1.0f
                        binding.btNext.isClickable = true
                        binding.btNext.isEnabled = true
                        binding.btNext.alpha = 1.0f
                    }
                }
            } else if (v.id == R.id.dusterIB) {
                mPaint?.strokeWidth = 40f
                mPaint?.xfermode = PorterDuffXfermode(PorterDuff.Mode.CLEAR)
            } else if (v.id == R.id.chalkIB) {
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel2, mPaint)
            } else if (v.id == R.id.backBT) {
                finish()
            }
        }
    }
}